frontend/catalyst/jax_tracer.py

```diff
@@ -114,6 +108,20 @@ def mark_gradient_tracing(method: str):
         TRACING_GRADIENTS.pop()
 
 
+def _make_execution_config(qnode):
+    """Updates the execution_config object with information about execution. This is 
+    used in preprocess to determine what decomposition and validation is needed."""
+
+    if qnode:
+        _gradient_method = _in_gradient_tracing(qnode)
+    else:
+        _gradient_method = None
+
+    return qml.devices.ExecutionConfig(
+        gradient_method=_gradient_method,
+    )
```

Comment:
```suggestion
    execution_config = qml.devices.DefaultExecutionConfig
    execution_config.gradient_method = _gradient_method
    return execution_config
```

Reply:
I would suggest to explicitly update `DefaultExecutionConfig` since the API declares this entity.

Reply:
@lillian542 later I came to another question: what part of the code is responsible of creating it in pure-PL programs? I mean, creating this config here seems strange to me. I guess someone should have created such a config on the PL side, right? Should we find and use this instance rather than creating our own?

Reply:
Good question! Giving users more direct access to updating the `ExecutionConfig` is an idea that has been floated before, and if it happens we will have to update this, but that's not the case at the moment.

Currently in pure PL programs, it's created with information from the `qnode` in a function `_make_exeuction_config` function (with a bit more going on in it than the catalyst version), then passed to the device. So this mimics the current behaviour in PennyLane (which takes place in part of the execution pipeline that catalyst doesn't use). You can see the [PennyLane version here](https://github.com/PennyLaneAI/pennylane/blob/38b3e74e26ef7f4f758e1cb34d4db0a1fc679780/pennylane/workflow/qnode.py#L1049) if you're curious.

The device preprocessing might then further update it (the `default.qubit` preprocessing does this, for example), and then it's passed along to `execute`. So device-specific modifications can be made by the developer for a device in preprocess. But users who aren't developing their own device never see the `ExecutionConfig`.

Reply:
I see, thanks!